core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeReservedValueManager.kt

```diff
@@ -0,0 +1,406 @@
+import org.hisp.dhis.android.core.arch.db.querybuilders.internal.WhereClauseBuilder
+import org.hisp.dhis.android.core.arch.helpers.UidsHelper.getUidOrNull
+import org.hisp.dhis.android.core.arch.repositories.scope.RepositoryScope
+import org.hisp.dhis.android.core.arch.repositories.scope.internal.RepositoryScopeOrderByItem
+import org.hisp.dhis.android.core.common.CoreColumns
+import org.hisp.dhis.android.core.common.IdentifiableColumns
+import org.hisp.dhis.android.core.maintenance.D2Error
+import org.hisp.dhis.android.core.maintenance.D2ErrorCode
+import org.hisp.dhis.android.core.maintenance.D2ErrorComponent
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnitProgramLinkTableInfo
+import org.hisp.dhis.android.core.organisationunit.internal.OrganisationUnitProgramLinkStore
+import org.hisp.dhis.android.core.organisationunit.internal.OrganisationUnitStore
+import org.hisp.dhis.android.core.program.ProgramTrackedEntityAttributeTableInfo
+import org.hisp.dhis.android.core.program.internal.ProgramTrackedEntityAttributeStore
+import org.hisp.dhis.android.core.settings.GeneralSettingObjectRepository
+import org.hisp.dhis.android.core.trackedentity.internal.ReservedValueSettingStore
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeReservedValueQuery
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeReservedValueStore
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeStore
+import org.hisp.dhis.android.core.user.internal.UserOrganisationUnitLinkStore
+
+@SuppressWarnings("LongParameterList", "TooManyFunctions")
+@Reusable
+class TrackedEntityAttributeReservedValueManager @Inject internal constructor(
+    private val store: TrackedEntityAttributeReservedValueStore,
+    private val organisationUnitStore: OrganisationUnitStore,
+    private val trackedEntityAttributeStore: TrackedEntityAttributeStore,
+    private val programTrackedEntityAttributeStore: ProgramTrackedEntityAttributeStore,
+    private val organisationUnitProgramLinkStore: OrganisationUnitProgramLinkStore,
+    private val userOrganisationUnitLinkStore: UserOrganisationUnitLinkStore,
+    private val generalSettingObjectRepository: GeneralSettingObjectRepository,
+    private val reservedValueSettingStore: ReservedValueSettingStore,
+    private val executor: D2CallExecutor,
+    private val reservedValueQueryCallFactory: QueryCallFactory<TrackedEntityAttributeReservedValue,
+        TrackedEntityAttributeReservedValueQuery>
+) {
+    private val d2ProgressManager = D2ProgressManager(null)
+
+    /**
+     * @param attributeUid        Attribute uid
+     * @param organisationUnitUid Organisation unit uid
+     * @return Value of tracked entity attribute
+     * @see .getValue
+     */
+    suspend fun blockingGetValue(attributeUid: String, organisationUnitUid: String): String {
```

Comment:
Public methods must not change the to `suspend` function because they are used by the app, sorry if it wasn't clear. The SDK must not change its public API at this level for 1.9.0.